tests/generic_config_updater/test_bgp_speaker.py

```diff
@@ -0,0 +1,197 @@
+            "Failed to update bgp speaker src address."
+        )
+        pytest_assert(re.search(BGPSPEAKER_IP_RANGE_RE.format(bgp_speaker, vlan_intf_ip_range), bgp_config),
+            "Failed to add bgp speaker ip range."
+        )
+
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+def bgp_speaker_tc1_add_dummy_ip_range(duthost, bgp_speaker, dummy_ip_range):
+    """ Test to add dummy ip range to existed config
+    """
+    json_patch = [
+        {
+            "op": "add",
+            "path": "/BGP_PEER_RANGE/{}/ip_range/1".format(bgp_speaker),
+            "value": "{}".format(dummy_ip_range)
+        }
+    ]
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+
+        bgp_config = show_bgp_running_config(duthost)
+        pytest_assert(re.search(BGPSPEAKER_IP_RANGE_RE.format(bgp_speaker, dummy_ip_range), bgp_config),
+            "Failed to add bgp speaker dummy ip range."
+        )
+
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+def bgp_speaker_tc1_rm_dummy_ip_range(duthost, bgp_speaker, dummy_ip_range):
+    """ Test to remove dummy ip range to existed config
+    """
+    json_patch = [
+        {
+            "op": "remove",
+            "path": "/BGP_PEER_RANGE/{}/ip_range/1".format(bgp_speaker)
+        }
+    ]
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+
+        bgp_config = show_bgp_running_config(duthost)
+        pytest_assert(not re.search(BGPSPEAKER_IP_RANGE_RE.format(bgp_speaker, dummy_ip_range), bgp_config),
+            "Failed to remove bgp speaker dummy ip range."
+        )
+
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+@pytest.mark.parametrize("ip_version", ["v4", "v6"])
+def test_bgp_speaker_tc1_test_config(duthost, lo_intf_ips, vlan_intf_ip_ranges, ip_version):
+    """ Test suite for bgp speaker config for v4 and v6
+    """
+    lo_ip, lo_ipv6 = lo_intf_ips
+    ip_range, ip_rangev6 = vlan_intf_ip_ranges
+
+    if ip_version == "v4":
+        lo_intf_ip = lo_ip
+        vlan_intf_ip_range = ip_range
+        bgp_speaker = BGPSPEAKER_V4
+        dummy_ip_range = DUMMY_IP_RANGE_V4
+    elif ip_version == "v6":
+        lo_intf_ip = lo_ipv6
+        vlan_intf_ip_range = ip_rangev6
+        bgp_speaker = BGPSPEAKER_V6
+        dummy_ip_range = DUMMY_IP_RANGE_V6
+    else:
+        pytest.fail("Invalid ip version!")
+
+    bgp_speaker_config_cleanup(duthost)
+    bgp_speaker_tc1_add_config(duthost, lo_intf_ip, vlan_intf_ip_range, bgp_speaker)
+    bgp_speaker_tc1_add_dummy_ip_range(duthost, bgp_speaker, dummy_ip_range)
+    bgp_speaker_tc1_rm_dummy_ip_range(duthost, bgp_speaker, dummy_ip_range)
```

Comment:
Can we add a test for changing the `src_address` switching from loopback0 to loobpack1?

Also, we would like to test adding a new VLAN and also a new Loopback is that covered under VLAN/Loopback testing? (i think so, just double checking).

Reply:
Added src_address change to a dummy ip.

> Also, we would like to test adding a new VLAN and also a new Loopback is that covered under VLAN/Loopback testing? (i think so, just double checking).

Yes. That is covered.